.github/workflows/upgrade_test_constraints.yml

```diff
@@ -37,17 +37,25 @@ jobs:
           python-version: "3.10"
           cache: pip
           cache-dependency-path: 'setup.cfg'
+      - uses: actions/setup-python@v4
+        with:
+          python-version: "3.11"
+          cache: pip
+          cache-dependency-path: 'setup.cfg'
+
       - name: Upgrade Python dependencies
         # ADD YOUR CUSTOM DEPENDENCY UPGRADE COMMANDS BELOW
         run: |
           python3.8 -m pip install -U pip pip-tools
           python3.9 -m pip install -U pip pip-tools
           python3.10 -m pip install -U pip pip-tools
+          python3.11 -m pip install -U pip pip-tools
 
           python3.8 -m piptools compile --upgrade -o resources/constraints/constraints_py3.8.txt setup.cfg resources/constraints/version_denylist.txt --extra pyqt5 --extra pyqt6_experimental --extra pyside2 --extra pyside6_experimental --extra testing --allow-unsafe --strip-extras --resolver=backtracking
           python3.9 -m piptools compile --upgrade -o resources/constraints/constraints_py3.9.txt setup.cfg resources/constraints/version_denylist.txt --extra pyqt5 --extra pyqt6_experimental --extra pyside2 --extra pyside6_experimental --extra testing --allow-unsafe --strip-extras --resolver=backtracking
           python3.9 -m piptools compile --upgrade -o resources/constraints/constraints_py3.9_examples.txt setup.cfg resources/constraints/version_denylist.txt resources/constraints/version_denylist_examples.txt --extra pyqt5 --extra pyqt6_experimental --extra pyside2 --extra pyside6_experimental --extra testing --allow-unsafe --strip-extras --resolver=backtracking
           python3.10 -m piptools compile --upgrade -o resources/constraints/constraints_py3.10.txt setup.cfg resources/constraints/version_denylist.txt --extra pyqt5 --extra pyqt6_experimental --extra pyside2 --extra pyside6_experimental --extra testing --allow-unsafe --strip-extras --resolver=backtracking
+          python3.11 -m piptools compile --upgrade -o resources/constraints/constraints_py3.11.txt setup.cfg resources/constraints/version_denylist.txt --extra pyqt5 --extra pyqt6_experimental --extra pyside2 --extra pyside6_experimental --extra testing --allow-unsafe --strip-extras --resolver=backtracking
```

Comment:
Side question, do anyone belive it would be worth replacing this by something like

```
for pyv in 3.8 3.9 3.10 3.11; do
      python${pyv}  -m pip install -U pip pip-tools
      python${pyv}  -m piptools compile --upgrade -o resources/constraints/constraints_${pyv}.txt  setup.cfg resources/constraints/version_denylist.txt --extra pyqt5 --extra pyqt6_experimental --extra pyside2 --extra pyside6_experimental --extra testing --allow-unsafe --strip-extras --resolver=backtracking
done
```

And have the `py3.9_examples.txt` on its own line ?

Reply:
I like it.